File: source/TiketKonser.java
public class TiketKonser {

    private final String kodeTiket;
    private final String namaEvent;
    private final double hargaTiket;
    private int kuotaTersedia;

    public TiketKonser(String kodeTiket, String namaEvent, double hargaTiket, int kuotaAwal) {
        if (hargaTiket <= 0) {
            throw new IllegalArgumentException("Harga tiket harus lebih besar dari nol");
        }
        if (kuotaAwal < 0) {
            throw new IllegalArgumentException("Kuota awal tidak boleh negatif");
        }
        this.kodeTiket = kodeTiket;
        this.namaEvent = namaEvent;
        this.hargaTiket = hargaTiket;
        this.kuotaTersedia = kuotaAwal;
    }

    public void jual(int jumlah) {
        if (jumlah <= 0) {
            throw new IllegalArgumentException("Jumlah tiket yang dijual harus lebih besar dari nol");
        }
        if (jumlah > kuotaTersedia) {
            throw new IllegalArgumentException("Kuota tidak mencukupi, sisa kuota: " + kuotaTersedia);
        }
        this.kuotaTersedia -= jumlah;
    }

    public int getKuotaTersedia() {
        return kuotaTersedia;
    }

    public double getHargaTiket() {
        return hargaTiket;
    }

    public void tampilkan() {
        System.out.println("Kode Tiket   : " + kodeTiket);
        System.out.println("Nama Event   : " + namaEvent);
        System.out.println("Harga Tiket  : Rp" + String.format("%,.0f", hargaTiket));
        System.out.println("Kuota Tersisa: " + kuotaTersedia);
    }
}

File: source/TiketKonser.php
<?php

declare(strict_types=1);

class TiketKonser
{
    public function __construct(
        private readonly string $kodeTiket,
        private readonly string $namaEvent,
        private readonly float $hargaTiket,
        private int $kuotaTersedia
    ) {
        if ($hargaTiket <= 0) {
            throw new InvalidArgumentException('Harga tiket harus lebih besar dari nol');
        }
        if ($kuotaTersedia < 0) {
            throw new InvalidArgumentException('Kuota awal tidak boleh negatif');
        }
    }

    public function jual(int $jumlah): void
    {
        if ($jumlah <= 0) {
            throw new InvalidArgumentException('Jumlah tiket yang dijual harus lebih besar dari nol');
        }
        if ($jumlah > $this->kuotaTersedia) {
            throw new InvalidArgumentException("Kuota tidak mencukupi, sisa kuota: {$this->kuotaTersedia}");
        }
        $this->kuotaTersedia -= $jumlah;
    }

    public function getKuotaTersedia(): int
    {
        return $this->kuotaTersedia;
    }

    public function getHargaTiket(): float
    {
        return $this->hargaTiket;
    }

    public function tampilkan(): void
    {
        $hargaFormatted = number_format($this->hargaTiket, 0, ',', '.');
        echo "Kode Tiket   : {$this->kodeTiket}\n";
        echo "Nama Event   : {$this->namaEvent}\n";
        echo "Harga Tiket  : Rp{$hargaFormatted}\n";
        echo "Kuota Tersisa: {$this->kuotaTersedia}\n";
    }
}
